src/Materials/Textures/texture.ts

```diff
@@ -701,7 +701,9 @@ export class Texture extends BaseTexture {
      */
     public static CreateFromBase64String(data: string, name: string, scene: Scene, noMipmap?: boolean, invertY?: boolean, samplingMode: number = Texture.TRILINEAR_SAMPLINGMODE,
         onLoad: Nullable<() => void> = null, onError: Nullable<() => void> = null, format: number = Constants.TEXTUREFORMAT_RGBA): Texture {
-        return new Texture("data:" + name, scene, noMipmap, invertY, samplingMode, onLoad, onError, data, false, format);
+        var texture = new Texture("data:" + name, scene, noMipmap, invertY, samplingMode, onLoad, onError, data, false, format);
+        scene.getEngine()._internalTexturesCache.push(texture.getInternalTexture()!);

```

Comment:
Why the engine is not caching the texture? because of "base64" ? In this case I would expect engine.createTexture to do the caching

Reply:
Yep, this is weird after debugging a bit more I found that this is already in the list so this line adds it a second time. Something else is removing it from the list before contextRestore is called.

Reply:
ok :)